src/arrayUtils.js

```diff
@@ -0,0 +1,68 @@
+if (typeof maybeExport === 'undefined') {
```

Comment:
Cool, makes sense to me. As far as processing the exports, it looks like `processMacros` is all that's needed. I'll take care of this now.